lib/server/server.mjs

```diff
@@ -0,0 +1,406 @@
+/* eslint-disable no-unused-vars */
+import stream from 'node:stream'
+import { Message, MessageError } from '../protocol.mjs'
+/* eslint-enable no-unused-vars */
+
+import assert from 'assert'
+import ndjson from 'ndjson'
+import pino from 'pino'
+import { nanoid } from 'nanoid'
+import { Request } from './request.mjs'
+import { Response } from './response.mjs'
+import { InvalidMessageError, StreamingError } from './error.mjs'
+
+/**
+* @summary Request handler callback.
+*
+* @description Request handlers are callbacks responding to a single incoming
+* request by inspecting the `request` and using the `response` to stream data as
+* appropriate.
+*
+* A single request may go through multiple request handlers. To enable data
+* sharing between request handlers, a `context` object is provided that can be
+* freely modified.
+*
+* For example, the first handler may take the `authorization` string from the
+* message header, look up the corresponding user from the DB and store the data
+* in the `context` object. Then, the next handler can perform operations on the
+* user, without needing to know how the user was authenticated.
+*
+* @callback RequestHandler
+* @param {Request} request Request data
+* @param {Response} response Response instance
+* @param {object} context Context object
+* @typedef {function(Request, Response, object)} RequestHandler
+*/
```

Comment:
I was expecting some function definition or something here.

Reply:
This is just the function type definition for JSDoc ( and LSP's for autocomplete and such ).

Reply:
Oh gotcha, sorry, my bad. :)